### ISSUE TITLE
feature request: provider/google/google_storage_bucket lifecycle interface

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-google#84]
_This issue was originally opened by @AVVS as hashicorp/terraform#13882. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

This is a feature request to support lifecycle part of the buckets

### Terraform Version

0.9/3

### Affected Resource(s)

- google_storage_bucket

### Terraform Configuration Files

```hcl
resource "google_storage_bucket" "permanent" {
  name     = "${module.common.buckets.permanent}"
  location = "us-west1"
  storage_class = "regional"

  lifecycle {
     action = {
       type = "Delete"
     }
     condition = {
       age = 1
     }
  }
}
```

### References
https://cloud.google.com/storage/docs/json_api/v1/buckets

**Comment by rileykarson** [comment]
Linking #6

**Comment by paddycarver** [comment]
This was resolved by #6, which was merged, and will be in the next release.
